src/components/SelectV2/Select.tsx

```diff
@@ -204,7 +204,7 @@ export const Select = ({
       >
         {variant !== 'default' && currentOption.startAdornment}{' '}
         {!variant.startsWith('short') &&
-          (currentOption.label || currentOption.value)}
+          (currentOption?.label || currentOption.value)}
```

Comment:
This one can be undefined, but has the same issue, might be better to default the currentOption to an empty object.

```ts
const currentOption = options.find(
    option => option.value === value
  ) || { } as OptionProps
```

Something like this.

Reply:
Ah yes this approach makes more sense, thanks!